modules/orchestrator/orchestrator-core/src/main/java/org/apache/airavata/orchestrator/cpi/impl/SimpleOrchestratorImpl.java

```diff
@@ -406,11 +406,17 @@ private String getTaskDag(List<String> taskIdList) {
                 DataType type = processOutput.getType();
                 switch (type) {
                     case STDOUT :
-                        processOutput.setValue(appName + ".stdout");
+                        if(null == processOutput.getValue() ||
+                                (null != processOutput.getValue() && processOutput.getValue().trim().isEmpty())){
```

Comment:
You can simplify this logic to (processOutput.getValue == null || processOutput.getvalue().trim().isEmpty())